pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -76,6 +85,50 @@ func (d ChangeSetWithVersionedRs) Calculate() ([]Change, error) {
 	return allChanges, nil
 }
 
+func (d ChangeSetWithVersionedRs) checkForMaxDurationAnn(existingRsGrouped map[string][]ctlres.Resource,
+	existingNonVerRsGrouped map[string]ctlres.Resource) error {
+
+	var err error
+	for _, res := range d.newRs {
+		if _, found := res.Annotations()[MaxDurationAnnKey]; found {
+			resKey := VersionedResource{res, nil}.UniqVersionedKey().String()
+			if exRes, found := existingNonVerRsGrouped[resKey]; found {
+				err = updateLastRenewedTime(res, exRes)
+			} else if exRes, found := existingRsGrouped[resKey]; found {
+				err = updateLastRenewedTime(res, exRes[len(exRes)-1])
```

Comment:
nit: adding a comment will make it more clear why we are doing exRes[len(exRes)-1]